### ISSUE TITLE
[C - Windows (cl)] VS Code integrated debug console is not working properly for I/O

### ISSUE THREAD

**Issue by malyjak** [microsoft/vscode-cpptools#6438]
While debugging C program on Windows 10 with `"externalConsole": false` set in `launch.json` all output must be finished with `fflush(stdout)` even when containing '\n' character in the end of `printf` function. Otherwise, the output is shown in the debug console only after program terminates. For input the problem is even worse. Using for example `fgets` function, the console will wait for input. However once something is on the input and enter is pressed ('\n' which should trigger the end of `fgets`), nothing happens. Both of these problems can be fixed using `"externalConsole": true`. However, it is very unpleasant in my case (many targets communicating between each other). I am drowning in external consoles.

**Comment by WardenGnaw** [comment]
> All output must be finished with fflush(stdout) even when containing '\n' character in the end of printf function. Otherwise, the output is shown in the debug console only after program terminates. 

This is by the language's design. printf writes to output buffer which is waiting to be flushed.
If no flushing occurs nothing will print until a flush is triggered which is via calling fflush or `\n`. The end of the program calls `exit()` which flushes the output buffer.

>  For input the problem is even worse. Using for example fgets function, the console will wait for input. However once something is on the input and enter is pressed ('\n' which should trigger the end of fgets), nothing happens.

<del>I can not reproduce this issue. Can you share your project + your launch.json/tasks.json?</del>
Sorry misread the title, I can reproduce this and we are working on improving the integrated terminal support for cppvsdbg.

**Comment by malyjak** [comment]
[io-demo.zip](https://github.com/microsoft/vscode-cpptools/files/5505112/io-demo.zip)
Exactly. If a char array contains '\n' it should be flushed by standard, but it is not in internal debugger window. I prepared some test code which should help demonstrate these issues. I also enclose some screenshots describing the problems. Please note for example how debugger variables and even highlighted debugger code line disappears, when debugger enter `fgets` function in internal console scenario. This is very strange behavior. The debugger never leaves this function even when input is presented. Also please note how all output is flushed only after main function terminates. Not when it should be due to '\n'. Hope this helps.

**Comment by sean-mcmanus** [comment]
@WardenGnaw So should the "by design" label be removed. Was there another issue tracking this already?

**Comment by WardenGnaw** [comment]
Duplicate of https://github.com/microsoft/vscode-cpptools/issues/5386
